let handleErrorByThrowingException() = ()

let handleErrorByReturningOption input = ???

let handleErrorByReturningResult input = ???

let bind switchFunction twoTrackInput = ???

let cleanupDisposablesWhenThrowingException resource = ???